source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+=========================
+Sensitive Data Encryption
+=========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Introduction
+------------
+
+Cras sed rutrum nisi, vitae tristique risus. Aliquam erat volutpat.
+Donec auctor nisl eget ornare accumsan. Pellentesque lectus tortor,
+consectetur molestie pellentesque ac, blandit non libero.
+
+Problem
+~~~~~~~
+
+Pellentesque non mi et est congue mattis pretium sit amet mi. Praesent
+nec tortor nec arcu molestie dapibus. Phasellus at nunc ut quam molestie
+finibus. Suspendisse finibus lectus eget nulla eleifend, vel consequat
+leo rhoncus. Phasellus lacinia tortor id aliquet condimentum. Morbi
+sodales est a leo rhoncus, non pretium neque dignissim.
+
+Solution
+~~~~~~~~
+
+Praesent sit amet elit pretium mi molestie porta a sit amet ipsum. Sed
+sollicitudin purus et mi blandit sodales.
+
+- Deterministic vs Non-Deterministic
+
+Procedure
+---------
+
+Requirements
+~~~~~~~~~~~~
+
+- Point to setup and install directions
+- Nullam imperdiet lorem vitae vulputate lacinia.
+- Donec eget velit tincidunt, gravida diam ac, efficitur lacus.
+
+A. Create a Local Master Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Aenean eu consequat lorem. Ut posuere est sed sodales pharetra. Cras
+volutpat, massa laoreet varius dictum, leo odio porttitor ante, nec
+auctor tortor orci et mi. Maecenas tempor, lacus vehicula molestie
+pulvinar, ante eros faucibus odio, sed consequat quam tellus vel arcu.
+Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere
+cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+
+B. Define a JSON Schema
+~~~~~~~~~~~~~~~~~~~~~~~
+
+MongoDB uses JSON Schema to to validate data which is written to a collection.
+Within the context of Field Level Encryption, we use JSON Schema to specify several
+configuration details. For instance, we can specify the form of encryption we want to use 
+- deterministic or random. We can also specify the BSON Type of the field we are encrypting. 
+In addition, we can specify the key we want to use. This is particularly important, because
+it means that we can use different keys for different fields.
+
+In the following example, we will go over creation of a schema for a collection which represents patient data
+in a medical office. Some fields use deterministic encryption, and others use random (probabilistic encryption). 
+
+The structure for our schema will be as follows:
+
+.. code::
```

Comment:
It might tie the document together better if this was a "skeletal" form of the JSON schema output that you showed later. E.g.
```
{ 
  ...
  fullName: <String, required>,
  address: <Deterministically Encrypted String, required>,
...
}

Reply:
Just to help me update the "Introduction/Problem/Solution" section, can you describe the use case for encrypting the address? In our planning, we did not mention that as one of the fields we would encrypt so what I wrote in the intro was under the assumption that only SSN and Medical Records were being encrypted:
https://docs.google.com/document/d/16fdXj-wTLeBNqe4FUwfc6dv_D4C3k2u7YZb_k2vcgMY/edit

Reply:
This actually changes the scenario which is currently:

```
In this scenario, we secure sensitive data on a Medical Care Management System which stores patients' personal information, appointments, and vital sign measurement reports for a fictitious company, *MedcoMD*. None of the patient data is public, and certain data such as their social security number (SSN) and vital sign measurements are particularly sensitive. It is important for the company and the patient that the data is kept private and secure.

MedcoMD needs this system to satisfy the following use cases:

- Doctors use the system to access Patients' medical records and add new vital sign measurements.
- Receptionists use the system to schedule appointments for Patients, who are identified by a combination of contact information and their SSN.
- Receptionists cannot access nor update Patients' medical records.

MedcoMD is also concerned with disclosure of private information through any of the following methods:

- Printing the information or leaving it displayed on a screen where unauthorized people can view it
- Communicating on an insecure network
- Gaining access to the database on one of the permissioned users

What can MedcoMD do to balance the functionality and access restrictions of their Medical Care Management System?```

Let's get together with Nick and figure out the best schema to use that fits a reasonable scenario as this is currently blocking work on the intro.